lib/authentication/auth_okta.js

```diff
@@ -164,13 +164,20 @@ function AuthOkta(connectionConfig, httpClient) {
   *
   * @returns {null}
   */
-  function validateURLs(authenticator, ssoUrl, tokenUrl) {
-    authenticator = authenticator.toLowerCase();
-    if (!(authenticator.startsWith(ssoUrl.substring(0, authenticator.length)) &&
-      authenticator.startsWith(tokenUrl.substring(0, authenticator.length)))) {
-      throw new Error('The prefix of the SSO/token URL and the specified authenticator do not match.');
+  this.validateURLs = function (authenticator, ssoUrl, tokenUrl) {
+    try {
+      const aUrl = new URL(authenticator);
+      const sUrl = new URL(ssoUrl);
+      const tUrl = new URL(tokenUrl);
+      if (!(compareUrlsByProtocolAndHost(aUrl, sUrl) && compareUrlsByProtocolAndHost(aUrl, tUrl))) {
+        throw new Error('The prefix of the SSO/token URL and the specified authenticator do not match.');
+      }
+    } catch (err) {
+      throw new Error('Authenticator, SSO, or token URL is invalid.');
     }
-  }
+  };
+
+  const compareUrlsByProtocolAndHost =  (firstUrl, secondUrl) => firstUrl.protocol === secondUrl.protocol && firstUrl.host === secondUrl.host;
```

Comment:
could we make this lambda a local variable in `validateURLs `?

Reply:
ok, moved it inside `validateURLs`